packages/backend/src/document.rs

```diff
@@ -95,24 +97,42 @@ pub async fn save_snapshot(state: AppState, data: RefContent) -> Result<(), AppE
     Ok(())
 }
 
-/// Gets an Automerge document ID for the document ref.
 pub async fn doc_id(state: AppState, ref_id: Uuid) -> Result<String, AppError> {
     let automerge_io = &state.automerge_io;
-    let ack = automerge_io.emit_with_ack::<Vec<Option<String>>>("get_doc", ref_id).unwrap();
-    let mut response = ack.await?;
-
-    let maybe_doc_id = response.data.pop().flatten();
-    if let Some(doc_id) = maybe_doc_id {
-        // If an Automerge doc handle for this ref already exists, return it.
-        Ok(doc_id)
-    } else {
-        // Otherwise, fetch the content from the database and create a new
-        // Automerge doc handle.
-        let content = head_snapshot(state.clone(), ref_id).await?;
-        let data = RefContent { ref_id, content };
-        let ack = automerge_io.emit_with_ack::<Vec<String>>("create_doc", data).unwrap();
-        let response = ack.await?;
-        Ok(response.data[0].to_string())
+
+    // Expecting an array of responses instead of a single response for unknowable reasons
```

Comment:
Haha, yes, when I writing the original code I wondered why `emit_with_ack` returns an array. I wish I knew.